mrpython/typechecking/typechecker.py

```diff
@@ -387,11 +390,20 @@ def type_check_FunctionDef(func_def, ctx):
     ctx.register_return_type(signature.ret_type)
     ctx.register_function_def(func_def, signature.partial)
 
-    # Step 3 : type-check body
+    # step 3 : type-check preconditions 
+
+    for precondition in func_def.preconditions:
+        precondition_type = type_expect(ctx, precondition, BoolType(), False)
+    if precondition_type is None:
```

Comment:
Isn't there a wrong indent ?  The precondition_type should be tested for all preconditions, no ?

Reply:
Oups, yes, that's a mistake, I'll correct that